app/src/main/java/org/thoughtcrime/securesms/migrations/LegacyMigrationJob.java

```diff
@@ -0,0 +1,305 @@
+ * (previously known as DatabaseUpgradeActivity). This job should *never* have new versions or
+ * migrations added to it. Instead, create a new {@link MigrationJob} and place it in
+ * {@link ApplicationMigrations}.
+ */
+public class LegacyMigrationJob extends MigrationJob {
+  
+  public static final String KEY = "LegacyMigrationJob";
+  
+  private static final String TAG = Log.tag(LegacyMigrationJob.class);
+
+  public  static final int NO_MORE_KEY_EXCHANGE_PREFIX_VERSION  = 46;
+  public  static final int MMS_BODY_VERSION                     = 46;
+  public  static final int TOFU_IDENTITIES_VERSION              = 50;
+  private static final int CURVE25519_VERSION                   = 63;
+  public  static final int ASYMMETRIC_MASTER_SECRET_FIX_VERSION = 73;
+  private static final int NO_V1_VERSION                        = 83;
+  private static final int SIGNED_PREKEY_VERSION                = 83;
+  private static final int NO_DECRYPT_QUEUE_VERSION             = 113;
+  private static final int PUSH_DECRYPT_SERIAL_ID_VERSION       = 131;
+  private static final int MIGRATE_SESSION_PLAINTEXT            = 136;
+  private static final int CONTACTS_ACCOUNT_VERSION             = 136;
+  private static final int MEDIA_DOWNLOAD_CONTROLS_VERSION      = 151;
+  private static final int REDPHONE_SUPPORT_VERSION             = 157;
+  private static final int NO_MORE_CANONICAL_DB_VERSION         = 276;
+  private static final int PROFILES                             = 289;
+  private static final int SCREENSHOTS                          = 300;
+  private static final int PERSISTENT_BLOBS                     = 317;
+  private static final int INTERNALIZE_CONTACTS                 = 317;
+  public  static final int SQLCIPHER                            = 334;
+  private static final int SQLCIPHER_COMPLETE                   = 352;
+  private static final int REMOVE_JOURNAL                       = 353;
+  private static final int REMOVE_CACHE                         = 354;
+  private static final int FULL_TEXT_SEARCH                     = 358;
+  private static final int BAD_IMPORT_CLEANUP                   = 373;
+  private static final int IMAGE_CACHE_CLEANUP                  = 406;
+  private static final int WORKMANAGER_MIGRATION                = 408;
+  private static final int COLOR_MIGRATION                      = 412;
+  private static final int UNIDENTIFIED_DELIVERY                = 422;
+  private static final int SIGNALING_KEY_DEPRECATION            = 447;
+  private static final int CONVERSATION_SEARCH                  = 455;
+
+
+  public LegacyMigrationJob() {
+    this(new Parameters.Builder().build());
+  }
+  
+  private LegacyMigrationJob(@NonNull Parameters parameters) {
+    super(parameters);
+  }
+
+  @Override
+  public boolean isUiBlocking() {
+    return true;
+  }
+
+  @Override
+  public @NonNull String getFactoryKey() {
+    return KEY;
+  }
+  
+  @Override
+  void performMigration() throws RetryLaterException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 41. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 108-254 into a separate method.